backend_tools/db_provider.ts

```diff
@@ -0,0 +1,122 @@
+import { doc, Firestore, getFirestore, setDoc, updateDoc, addDoc, getDoc, DocumentData, Query, where, getDocs, QueryDocumentSnapshot, query, QuerySnapshot, deleteDoc, collection, arrayUnion } from "firebase/firestore"
+import Club from "../models/club";
+import User from "../models/user";
+import FirebaseApp from "./firebase";
+import Event from "../models/event";
+
+
+
+class DbProvider {
+    db: Firestore = getFirestore(FirebaseApp)
+    user_path: string = "Users"
+    club_path: string = "Clubs"
+    event_path: string = "Events"
+    async createUser(user: User): Promise<boolean> {
+        try {
+
+
+            const userCollection = collection(this.db, this.user_path)
+            await addDoc(userCollection, user);
+            return true;
+        }
+        catch (error) {
+            console.error(`failure saving user, got error: ${error}`);
+            return false;
+        }
+    }
+
+    async createClub(club: Club): Promise<boolean> {
+        try {
+            //const clubCollection = collection(this.db, this.club_path)
+            const docu = doc(this.db, this.club_path, club.name)
+
+            await setDoc(docu, club)
+
+            return true;
+
+        } catch (error) {
+            console.error(`failure saving club, got error: ${error}`);
+            return false
+        }
+    }
+
+    async getUser(userid: string): Promise<User> {
+        const userReference = doc(this.db, this.user_path, userid)
+        const ref = await getDoc<DocumentData>(userReference)
+        return ref.data() as unknown as User;
+    }
+
+    async getClub(club_name: string): Promise<Club> {
+        const clubReference = doc(this.db, this.club_path, club_name)
+        const ref = await getDoc<DocumentData>(clubReference)
+        return ref.data() as Club;
+    }
+
+    async getSomeClubs(): Promise<Club[] | null> {
```

Comment:
Could this potentially be renamed to best indicate what it is doing, if I'm understanding correctly? 
```suggestion
    async getAllClubs(): Promise<Club[] | null> {
```

Reply:
You're right